opensearchutil/bulk_indexer.go

```diff
@@ -111,16 +109,37 @@ type BulkIndexerStats struct {
 // BulkIndexerItem represents an indexer item.
 //
 type BulkIndexerItem struct {
-	Index           string
-	Action          string
-	DocumentID      string
-	Body            io.Reader
-	RetryOnConflict *int
+	Index               string
+	Action              string
+	DocumentID          string
+	Routing             *string
+	Version             *int64
+	VersionType         *string
+	IfSeqNum            *int64
+	IfPrimaryTerm       *int64
+	WaitForActiveShards interface{}
```

Comment:
Why is this an interface{} as opposed to a string? From [here](https://opensearch.org/docs/opensearch/rest-api/document-apis/index-document/#url-parameters) it appears to be a string.

Reply:
since it can either be an integer or the string `all` I was thinking we could support either as an interface, no problem changing it to be just a string if that's what we'd like to do.